toc.md

```diff
@@ -1,54 +1,76 @@
 
-This is where background job processors come in. Background job processors run as a separate process decoupled from your application. As a developer of your application, you can delegate these time-consuming tasks to a background job processor to handle these time-consuming tasks while allowing the user to still interact with your website. These tasks are performed asynchronously when you have resources available.
+## Background Job Feature Overview
+
+Before we get into how to build a Background Job Processor, let's take a quick run at the features a BJP needs to have, based on research done on other popular BJPs:
 
-## Popular Features of BJPs
-Before we get into building a background job processor, let's first introduce some of the common features we see in other background job processors:
 ##### Core Features:
-These are features that all background job processors *should* have.
-* **asynchrony**: background job processors should handle a task in a separate process away from the main application, so that your main web application is free to continue handling more requests from the user.
-* **reliability**: background job processors should be able to handle any errors that occur gracefully.
-* **efficiency:** background job processors should perform their tasks in a timely manner so that queues do not get a backlog of things that still needs to be done.
-* **scalability**: background job processors should run fine and scale in the context of a distributed system, such as multiple web servers.
-* **reporting**: background job processors should track job statistics for information about background jobs, allowing for useful decision making.
-
-##### Bonus Features:
-These are features that are not necessary for background job processors, but these can be added for robustness.
-* **configurability**: allows a developer to tweak options that best fits their own application's needs.
-* **ease of use**: simple to use out-of-the-box with rails.
+* **asynchrony**: a BJP should handle a task in a different process than the one used for the main application, so that the main application is free to continue handling more requests
+* **reliability**: a BJP should be able to handle errors gracefully and be resilient in case of a crash
+* **efficiency:** a BJP should perform the jobs that it is handed in a timely manner so that it does not get a constantly increasing backlog of jobs
+* **scalability**: a BJP should scale in the context of a distributed system
+* **reporting**: a BJP should track statistics about jobs, errors and other related information for better decision making
+
+##### Additional Features:
+* **configurability**: a BJP should be configurable in order to allow developers to tweak options so that the BJP can best fit their own application's requirements
+* **ease of use**: a BJP should be simple to use out-of-the-box and integrate with rails in the context of building it in Ruby
```

Comment:
I think it's hard to understand what we mean when we say "and integrate with rails in the context of building it in Ruby"
Might be better to say something like "and integrate with Rails (since it's been written in Ruby)"